integrations/azure_ai_search/src/haystack_integrations/document_stores/azure_ai_search/document_store.py

```diff
@@ -0,0 +1,414 @@
+        keys_to_remove = ["dataframe", "blob", "sparse_embedding", "score"]
+        index_document = {k: v for k, v in document.items() if k not in keys_to_remove}
+        metadata = index_document.pop("meta", None)
+        for key, value in metadata.items():
+            index_document[key] = value
+        if index_document["embedding"] is None:
+            index_document["embedding"] = self._dummy_vector
+
+        return index_document
+
+    def _create_metadata_index_fields(self, metadata: Dict[str, Any]) -> List[SimpleField]:
+        """Create a list of index fields for storing metadata values."""
+
+        index_fields = []
+        metadata_field_mapping = self._map_metadata_field_types(metadata)
+
+        for key, field_type in metadata_field_mapping.items():
+            index_fields.append(SimpleField(name=key, type=field_type, filterable=True))
+
+        return index_fields
+
+    def _map_metadata_field_types(self, metadata: Dict[str, type]) -> Dict[str, str]:
+        """Map metadata field types to Azure Search field types."""
+        metadata_field_mapping = {}
+
+        for key, value_type in metadata.items():
+            field_type = type_mapping.get(value_type)
+            if not field_type:
+                error_message = f"Unsupported field type for key '{key}': {value_type}"
+                raise ValueError(error_message)
+            metadata_field_mapping[key] = field_type
+
+        return metadata_field_mapping
+
+    def _embedding_retrieval(
+        self,
+        query_embedding: List[float],
+        *,
+        top_k: int = 10,
+        fields: Optional[List[str]] = None,
+        filters: Optional[Dict[str, Any]] = None,  # TODO will be used in the future
+    ) -> List[Document]:
+        """
+        Retrieves documents that are most similar to the query embedding using a vector similarity metric.
+        It uses the vector configuration of the document store. By default it uses the HNSW algorithm with cosine similarity.
+
+        This method is not meant to be part of the public interface of
+        `AzureAISearchDocumentStore` nor called directly.
+        `AzureAISearchEmbeddingRetriever` uses this method directly and is the public interface for it.
+
+        :param query_embedding: Embedding of the query.
+        :param filters: Filters applied to the retrieved Documents. Defaults to None.
+            Filters are applied during the approximate kNN search to ensure that top_k matching documents are returned.
+        :param top_k: Maximum number of Documents to return, defaults to 10
+
+        :raises ValueError: If `query_embedding` is an empty list
+        :returns: List of Document that are most similar to `query_embedding`
+        """
+
+        if not query_embedding:
+            msg = "query_embedding must be a non-empty list of floats"
+            raise ValueError(msg)
+
+        vector_query = VectorizedQuery(vector=query_embedding, k_nearest_neighbors=3, fields="embedding")
```

Comment:
During testing, I noticed that this value overrides the `top_k` parameter that is passed in on the next line, so the line should be:
```
vector_query = VectorizedQuery(vector=query_embedding, k_nearest_neighbors=top_k, fields="embedding")
```

Reply:
This is added in the next commit. 🙏